src/pages/accommodations/AccommodationForm.tsx

```diff
@@ -0,0 +1,286 @@
+import { zodResolver } from '@hookform/resolvers/zod';
+import { Box, Button, Stack, TextField, Typography } from '@mui/material';
+import { DateField, LocalizationProvider } from '@mui/x-date-pickers';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import dayjs from 'dayjs';
+import { Controller, useForm } from 'react-hook-form';
+import { useNavigate, useParams } from 'react-router-dom';
+
+import { useCreateAccommodation } from '@/api/mutations/accommodations/useCreateAccommodation';
+import { useDeleteAccommodation } from '@/api/mutations/accommodations/useDeleteAccommodation';
+import { useUpdateAccommodation } from '@/api/mutations/accommodations/useUpdateAccommodation';
+import { useGetAccommodation } from '@/api/queries/accommodations/useGetAccommodation';
+import { ROUTES } from '@/config/routes.config';
+import { AccommodationReq, accommodationSchema } from '@/types/accommodation.types';
+
+export default function AccommodationForm() {
+  const { id } = useParams();
+  const navigate = useNavigate();
+  const { data: accommodation } = useGetAccommodation({ id });
+  const { mutate: createAccommodation } = useCreateAccommodation();
+  const { mutate: updateAccommodation } = useUpdateAccommodation();
+  const { mutate: deleteAccommodation } = useDeleteAccommodation();
+
+  const {
+    control,
+    handleSubmit,
+    formState: { errors, dirtyFields },
+  } = useForm<AccommodationReq>({
+    resolver: zodResolver(accommodationSchema),
+    defaultValues: {
+      name: '',
+      description: '',
+      allowedNumberOfPeople: 1,
+      availableFrom: String(new Date()),
+      availableTo: String(new Date(dayjs().add(1, 'week').toDate())),
+      thumbnailUrl: '',
+      numberOfRooms: 1,
+      previewImgUrl: '',
+      price: 0,
+      squareMeters: 0,
+      address: {
+        street: '',
+        city: '',
+        country: '',
+        zipCode: '',
+        latitude: 0,
+        longitude: 0,
+      },
+    },
+    values: accommodation?.data,
+  });
+
+  const onSubmit = (data: AccommodationReq) => {
+    if (id) {
+      if (!Object.keys(dirtyFields).length) {
+        navigate(ROUTES.accommodations.root);
+        return;
+      } else {
```

Comment:
you don't need 'else' if you user return in 'if' statement. remove 'else' from this function at all

Reply:
but if we remove else then it's gonna first updateAccommodation then also tries to createAccommodation

    if (id) { <- Checking if need to update or create
      if (!Object.keys(dirtyFields).length) {  <- Checking if values ever changed so we can avoid unnecessary request 
        navigate(ROUTES.accommodations.root);
        return;
      } else {
        updateAccommodation({
          ...data,
          id,
          availableFrom: dayjs(data.availableFrom).toISOString(),
          availableTo: dayjs(data.availableTo).toISOString(),
        });
      }
    } else {
      createAccommodation({
        ...data,
        availableFrom: dayjs(data.availableFrom).toISOString(),
        availableTo: dayjs(data.availableTo).toISOString(),
      });
    }